components/scream/src/physics/shoc/shoc_assumed_pdf_impl.hpp

```diff
@@ -263,11 +270,17 @@ void Functions<S,D>::shoc_assumed_pdf(
         std_s1 = ekat::sqrt(ekat::max(0,
                                       ekat::square(cthl1)*thl2_1
                                       + ekat::square(cqt1)*qw2_1 - 2*cthl1*sqrtthl2_1*cqt1*sqrtqw2_1*r_qwthl_1));
+        const auto std_s1_not_small = std_s1 > 2.0E-155;
```

Comment:
Here is the overflow problem on the C++ side. With very small std_s1, the division below can overflow max float.

Reply:
If possible, use a numeric_limits<Real> function so this threshold works for single precision too.